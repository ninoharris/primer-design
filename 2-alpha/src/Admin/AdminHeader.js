import React from 'react'
import { connect } from 'react-redux'
import { Link, withRouter } from 'react-router-dom'
import { startLogout } from '../actions/auth'


export const AdminHeader = ({ 
  title, 
  children,
  startLogout,
  location, 
  match, 
  ...rest 
}) => {
  return (
    <div className="row">
      <div className="Nav">
        <div className="Logo">
          <h4>{title}</h4>
        </div>
        <div>
          {children}
          {location.pathname !== '/admin/create' && !location.pathname.includes('/admin/edit') && 
          <Link to="/admin/create"><button className="btn btn-success mr-3">Add new exercise</button></Link>}
          <button className="btn btn-info mr-3" onClick={() => { }}>View students</button>
          {location.pathname !== '/admin' ? <Link to="/admin" className="btn btn-success Goto-Home">Back to home</Link> : ''}
          <button className="Logout-Button btn btn-warning mr-3" onClick={startLogout}>Log out</button>
        </div>
      </div>
    </div>
  )
}

const mapDispatchToProps = (dispatch) => {
  return {
    startLogout: () => dispatch(startLogout)
  }
}

export default withRouter(connect(null, mapDispatchToProps)(AdminHeader))