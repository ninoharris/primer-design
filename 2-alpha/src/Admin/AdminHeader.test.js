import React from 'react'
import { shallow } from 'enzyme'

import { AdminHeader } from './AdminHeader'

test('AdminHeader renders correctly', () => {
  
  const wrapper = shallow(<AdminHeader startLogout={() => {}} location={{pathname: '/'}} />)
  expect(wrapper).toMatchSnapshot()
})

test('AdminHeader calls startLogout function when logout button is clicked', () => {
  const startLogout = jest.fn()
  const wrapper = shallow(<AdminHeader startLogout={startLogout} location={{ pathname: ''}} />)
  expect(startLogout.mock.calls.length).toBe(0)
  wrapper.find('.Logout-Button').simulate('click')
  expect(startLogout.mock.calls.length).toBe(1)
})

describe('AdminHeader home button display', () => {
  test('Home button is hidden for exactly /admin', () => {
    const wrapper = shallow(<AdminHeader startLogout={() => {}} location={{pathname: '/admin'}} />)
    expect(wrapper.find('.Goto-Home').length).toBe(0)
  })
  test('Home button is hidden for any path that is not /admin', () => {
    const wrapper = shallow(<AdminHeader startLogout={() => { }} location={{ pathname: '/admin/null' }} />)
    expect(wrapper.find('.Goto-Home').length).toBe(1)
  })
})