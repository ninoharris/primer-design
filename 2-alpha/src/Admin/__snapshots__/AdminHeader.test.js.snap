// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHeader renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="Nav"
  >
    <div
      className="Logo"
    >
      <h4 />
    </div>
    <div>
      <Link
        replace={false}
        to="/admin/create"
      >
        <button
          className="btn btn-success mr-3"
        >
          Add new exercise
        </button>
      </Link>
      <Link
        className="btn btn-info mr-3"
        replace={false}
        to="/admin/cohorts"
      >
        View cohorts
      </Link>
      <Link
        className="btn btn-success Goto-Home"
        replace={false}
        to="/admin"
      >
        Back to home
      </Link>
      <span
        className="Author-Name"
      >
        administrator
      </span>
      <Link
        className="btn btn-info mr-3"
        replace={false}
        to="/admin/my-account"
      >
        My Account
      </Link>
      <button
        className="Logout-Button btn btn-warning mr-3"
        onClick={[Function]}
      >
        Log out
      </button>
    </div>
  </div>
</div>
`;
